cmd/tracee-ebpf/main.go

```diff
@@ -344,6 +262,122 @@ func main() {
 	}
 }
 
+func prepareBpfObject(config *tracee.Config, kConfig *helpers.KernelConfig, OSInfo *helpers.OSInfo) error {
+
+	var d = struct {
+		btfenv     bool
+		bpfenv     bool
+		btfvmlinux bool
+	}{
+		btfenv:     false,
+		bpfenv:     false,
+		btfvmlinux: helpers.OSBTFEnabled(),
+	}
+
+	bpfFilePath, err := checkEnvPath("TRACEE_BPF_FILE")
+	if bpfFilePath != "" {
+		d.bpfenv = true
+	} else if bpfFilePath == "" && err != nil {
+		return err
+	}
+	btfFilePath, err := checkEnvPath("TRACEE_BTF_FILE")
+	if btfFilePath != "" {
+		d.btfenv = true
+	} else if btfFilePath == "" && err != nil {
+		return err
+	}
+	if debug {
+		fmt.Printf("BTF: bpfenv = %v, btfenv = %v, vmlinux = %v\n", d.bpfenv, d.btfenv, d.btfvmlinux)
+	}
+
+	var bpfBytes []byte
+	var unpackBTFFile string
+
+	// Decision ordering:
```

Comment:
I think this flow can be greatly simplified as follows:

```
if bpfenv {
  config.bpf = bpfenv
} else {
  config.bpf =  unpack
}
if btfenv {
  config.btf = btfenv
} else if vmlinux {
  config.btf = vmlinux
}
```

WDYT?

Reply:
Definitely. I knew this had space for improvement, I'll improve it by using this suggestion.

Reply:
@danielpacak like we've spoken, would you like to simplify this and get into the BPF and BTF loading logic ?

Reply:
Sure, I've created #1275 to work on that.

Reply:
Thanks! Marking this as resolved (per new issue created).